Update to Marketplace and new buttons

JorgeVidal29 opened the pull request:
**Description**
New marketplace to make api connection easier, and new primary buttons

Resolves #17 and #29 passively. Main things were done in a previous PR, I made adjustments to make it work with production changes -Emil

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

Zexdur committed:
New Marketplace and Buttons

Updated Marketplace and made it up to par with review. New Buttons are also in the main css

Zexdur committed:
Updated for Flow

SecondThread commented on src/Main.css:
```diff
@@ -307,4 +307,106 @@ body {
+  color: #666666;
+}
+
+/*Select Styling*/
```
?? Unhelpful comment. I don't know what this code is for/does, but I knew it was select styling from the fact that it is for "select" in a Cascading Style Sheets file.

SecondThread commented on src/Main.css:
```diff
@@ -307,4 +307,106 @@ body {
+}
+
+/* Button Styling */
+.btn {
```
This would mean we have two definitions for the class .btn

I'm not a CSS expert, so I don't know if that is techincally allowed, but it confuses me, so I'm not a fan of it.

SecondThread commented on src/marketplace/AccessoriesView.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+class AccessoriesView extends React.Component<{||}> {
+    render()
+    {    return (
```
Wrong line for {

SecondThread commented on src/marketplace/CannonsView.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+class CannonsViews extends React.Component<{||}> {
+    render()
+    {    return (
```
ditto

SecondThread commented on src/marketplace/CasusBlocksView.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+class CasusBlocksView extends React.Component<{||}> {
+    render()
+    {    return (
```
you get the idea...

SecondThread commented on src/marketplace/Marketplace.js:
```diff
@@ -3,11 +3,61 @@
+		let partView='NONE';
+		switch(this.state.marketplaceViewClicked) {
+			case 'CHASSIS':
+				console.log("Here");
```
No debug output in prod please :)

SecondThread commented on src/marketplace/Marketplace.js:
```diff
@@ -3,11 +3,61 @@
+			case 'NONE':
+				partView=(<h2>Select a type on the left</h2>);
+				break;
+			}
```
This should line up with the switch statement, not the nested cases

SecondThread commented on src/marketplace/Marketplace.js:
```diff
@@ -21,31 +71,27 @@ class Marketplace extends React.Component<{||}> {
-						<h6 className="text-right">Tank One Jammer</h6>
-						<h6 className="text-right">Tank One Base</h6>
+						<div className="text-center">
+						{partView}
```
This should be indented one more layer, since it is inside a div block

SecondThread reviewed (changes requested):
See comments

blaiadam commented on src/marketplace/Marketplace.js:
```diff
@@ -21,31 +71,27 @@ class Marketplace extends React.Component<{||}> {
-		  			<div className="col-md-4">
-						<h2 className="text-center">Selected Tank</h2>
+					  <div className="list-group col-md-6">
+						<a className="list-group-item list-group-item-action disabled">Category</a>
```
All of the "a" tags generate an error for a broken link. If this is something that can't be fixed until later, then disregard this comment. Otherwise, please fix.

Here is one of the errors:

> The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with the appropriate styles.

SecondThread replied:
This is still broken, please address:
![image](https://user-images.githubusercontent.com/18317476/74443139-b2a12480-4e40-11ea-8ac0-d9d405ca9825.png)

blaiadam commented on src/marketplace/Marketplace.js:
```diff
@@ -3,11 +3,61 @@
+			}
 		return (
 	  		<div id="Parent">
 				<div className="row">
```
Add the Navbar somewhere around here.

blaiadam reviewed (changes requested):
Although I don't think it has to be done with this PR, we should implement your buttons to the rest of the pages. Also, if we are using your new buttons, we can remove the ones I setup before.

This looks like a good start. Just resolve David's concerns and add the Navbar to the top of your new page, instructions on how to use it are in the Navbar.js file. 

This fails Flow as well. I commented where Flow is having an issue.

blaiadam commented on src/marketplace/Marketplace.js:
```diff
@@ -3,11 +3,61 @@
+import TreadsView from './TreadsView.js';
+import AccessoriesView from './AccessoriesView.js';
+import ScannersView from './ScannersView.js';
+import MarketPlaceViewDataTypes from './MarketPlaceViewDataTypes.js';
```
The import for the MarketPlaceViewDataTypes is incorrect. Here is what it should be:
`import type { MarketPlaceViewDataTypes } from './MarketPlaceViewDataTypes.js';`

Zexdur committed:
Updated Marketplace and Added Signup API

Connected the login/signup api and have updated the marketplace to be up to coding standards

SecondThread commented on src/marketplace/AccessoriesView.js:
```diff
@@ -1,8 +1,8 @@
 class AccessoriesView extends React.Component<{||}> {
-    render()
-    {    return (
+    render(){    
```
Space between the close paren and open curly brace please.

SecondThread commented on src/login/LoginPopup.js:
```diff
@@ -36,22 +35,22 @@ class LoginPopup extends React.Component<Props, State> {
 	}
 
-	componentDidMount():void {
+	/*componentDidMount():void {
```
No dead code blocks in production please

SecondThread replied:
^

SecondThread commented on src/marketplace/JammersView.js:
```diff
@@ -1,8 +1,8 @@
 class JammersView extends React.Component<{||}> {
-    render()
-    {    return (
+    render(){    
```
ditto, space please

SecondThread commented on src/marketplace/Marketplace.js:
```diff
@@ -33,7 +34,6 @@ class Marketplace extends React.Component<Props, State> {
 		let partView='NONE';
 		switch(this.state.marketplaceViewClicked) {
```
Looks good thanks for fixing :)

SecondThread replied:
This generates a warning saying that a default case is expected.

SecondThread commented on src/marketplace/Marketplace.js:
```diff
@@ -57,18 +57,10 @@ class Marketplace extends React.Component<Props, State> {
+				<Navbar styleName="navbtn" linkName="Mainmenu" returnName="Back to Main Menu" pageName="Marketplace" userName="FRIcker | $465128"/>
 				<div className="row mt-5">
 		  			<div className="col-md-4">
 					  <div className="list-group col-md-6">
```
Indent line 87 (old) / 79 (new) containing `{partView}` since it is inside a div tag please

Derplime committed:
SignupPopup.js: modified signup fetch to new API route naming

Derplime committed:
LoginPopup.js: edited login click fetch to match new API route naming in production.

SecondThread commented on src/login/LoginPopup.js:
```diff
@@ -36,22 +35,22 @@ class LoginPopup extends React.Component<Props, State> {
 		
 		return body;
-	};
+	};*/
```
^ No dead code in prod please

SecondThread commented on src/marketplace/TreadsView.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+class TreadsView extends React.Component<{||}> {
+    render()
+    {    return (
```
plz fix formatting

SecondThread reviewed (changes requested):
Requested changes weren't fixed, please fix

JorgeVidal29 committed:
Merge branch 'master' into Jorge

Zexdur committed:
Update to marketplace

Zexdur committed:
Merge branch 'Jorge' of https://github.com/RickLeinecker/Fortuna into Jorge

SecondThread reviewed (changes requested):
Generates new warnings as mentioned by @blaiadam on Feb. 3

SecondThread commented:
Also it doesn't pass flow

Zexdur committed:
Update Marketplace

Zexdur committed:
Update index.js

SecondThread committed:
fixed capitolization of battlearena

blaiadam commented:
Passes flow on my end. Just fix the anchor errors that David brought up and I'll approve.

Zexdur committed:
Updated links to buttons

blaiadam commented:
Thank you for fixing the anchor references. The switch statement you have needs a default case. I think this is the last thing that we have mentioned that needs fixing.

![Screenshot (21)](https://user-images.githubusercontent.com/42626045/74581687-1d0cae80-4f80-11ea-9731-e472a35118cf.png)

SecondThread reviewed (approved):
Thanks for fixing
